index.html

```diff
@@ -361,48 +365,67 @@ <h3>
           rejected with=] with a {{"NotAllowedError"}} {{DOMException}}.
           </li>
-          <li data-tests="wakelock-document-hidden-manual.https.html">If
-          |document| is <a>hidden</a>, return [=a promise rejected with=]
-          {{"NotAllowedError"}} {{DOMException}}.
+          <li data-tests="wakelock-document-hidden-manual.https.html">If the
+          steps to <a>determine the visibility state</a> return `hidden`,
+          return [=a promise rejected with=] {{"NotAllowedError"}}
+          {{DOMException}}.
+          </li>
+          <li>Let |record| be the <a>platform wake lock</a>'s <a>state
+          record</a> associated with |document| and |type|.
           </li>
           <li>Let |promise:Promise| be [=a new promise=].
           </li>
-          <li>Return |promise| and run the following steps <a>in parallel</a>:
+          <li>Run the following steps <a>in parallel</a>:
             <ol>
-              <li>
-                <a>Abort when</a> |document| is <a>hidden</a>.
-              </li>
-              <li>Let |state:PermissionState| be the result of awaiting
-              <a>obtain permission</a> steps with "`screen-wake-lock`":
+              <li>Let |state:PermissionState| be the result of invoking
+              <a>obtain permission</a> with "`screen-wake-lock`":
                 <ol>
                   <li data-tests="wakelock-request-denied.https.html">If
-                  |state| is {{PermissionState/"denied"}}, then reject
-                  |promise| with a {{"NotAllowedError"}} {{DOMException}}, and
-                  abort these steps.
+                  |state| is {{PermissionState/"denied"}}, then <a>queue a
+                  global task</a> on the <a>screen wake lock task source</a>
+                  given |global| to reject |promise| with a
+                  {{"NotAllowedError"}} {{DOMException}}, and abort these
+                  steps.
                   </li>
                 </ol>
               </li>
-              <li>Let |lock:WakeLockSentinel| be a new {{WakeLockSentinel}}
-              object with its {{WakeLockSentinel/type}} attribute set to
-              |type|.
-              </li>
-              <li>Invoke <a>acquire a wake lock</a> with |lock| and
-              {{WakeLockType/"screen"}}.
-                <aside class="note">
-                  The <a>acquire a wake lock</a> algorithm may ultimately be
-                  unable to acquire a lock from the operating system, but this
-                  is indistinguishable from a successful lock acquisition to
-                  avoid user fingerprinting (failure to acquire a lock can
-                  indicate low battery levels, for example).
-                </aside>
-              </li>
-              <li>Resolve |promise| with |lock|.
+              <li>
+                <a>Queue a global task</a> on the <a>screen wake lock task
+                source</a> given |global| to run these steps:
```

Comment:
Not totally clear what "given |global|" means here.

Reply:
It's defined in the beginning of the steps as "`this's` relevant object", is there a clearer way to refer to it?

Reply:
domenic used wording like "given |document|'s [=relevant global object=] to"

Reply:
Yeah, I noticed shortly after posting the comment above :-D I've pushed a new version incorporating that.